src/fluree/server/system.clj

```diff
@@ -13,129 +27,619 @@
-  (let [config* (assoc config :ledger-directory (:storage-path server))]
-    (raft/start config*)))
+(defmethod ig/init-key :fluree.server.consensus/raft
+  [_ config]
+  (let [log-history      (get-first-value config log-history-iri)
+        entries-max      (get-first-value config entries-max-iri)
+        catch-up-rounds  (get-first-value config catch-up-rounds-iri)
+        servers          (get-values config raft-servers-iri)
+        this-server      (get-first-value config this-server-iri)
+        log-directory    (get-first-value config log-directory-iri)
+        ledger-directory (get-first-value config ledger-directory-iri)]
+    (raft/start {:log-history      log-history
+                 :entries-max      entries-max
+                 :catch-up-rounds  catch-up-rounds
+                 :servers          servers
+                 :this-server      this-server
+                 :log-directory    log-directory
+                 :ledger-directory ledger-directory})))
 
-(defmethod ig/halt-key! :fluree/raft
+(defmethod ig/halt-key! :fluree.server.consensus/raft
   [_ {:keys [close] :as _raft-group}]
   (close))
 
-(defmethod ig/init-key :fluree/standalone
-  [_ {:keys [conn subscriptions watcher max-pending-txns]}]
-  (standalone/start conn subscriptions watcher max-pending-txns))
+(defmethod ig/init-key :fluree.server.consensus/standalone
+  [_ {:keys [subscriptions watcher] :as config}]
+  (let [max-pending-txns (get-first-value config max-pending-txns-iri)
+        conn             (get-first config connection-iri)]
+    (standalone/start conn subscriptions watcher max-pending-txns)))
 
-(defmethod ig/halt-key! :fluree/standalone
+(defmethod ig/halt-key! :fluree.server.consensus/standalone
   [_ transactor]
   (standalone/stop transactor))
 
-(defmethod ig/init-key :fluree/handler
-  [_ {:keys [conn consensus watcher subscriptions root-identities closed-mode]}]
-  (handler/app conn consensus watcher subscriptions root-identities closed-mode))
+(defmethod ig/init-key :fluree.server.api/handler
+  [_ {:keys [connection consensus watcher subscriptions root-identities closed-mode]}]
+  (handler/app connection consensus watcher subscriptions root-identities closed-mode))
 
-(defmethod ig/init-key :http/jetty
-  [_ {:keys [handler port join?]}]
-  (jetty/run-jetty handler {:port port, :join? join?}))
+(defmethod ig/init-key :fluree.server.http/jetty
+  [_ {:keys [handler] :as config}]
+  (let [port (get-first-value config http-port-iri)]
+    (jetty/run-jetty handler {:port port, :join? false})))
 
-(defmethod ig/halt-key! :http/jetty
+(defmethod ig/halt-key! :fluree.server.http/jetty
   [_ http-server]
   (jetty/stop-server http-server))
 
-(defmethod ig/init-key :fluree/sid-migration
+(defmethod ig/init-key :fluree.server/sid-migration
   [_ {:keys [conn ledgers force]}]
   (task.migrate-sid/migrate conn ledgers force))
 
+(defmethod ig/init-key :default
+  [_ config]
+  config)
+
+(def default-resource-name "config.jsonld")
```

Comment:
@zonotope Right now it seems `config.jsonld` is an example of what's possible (e.g. with the s3 & ipns publishers). But since `resources/config.jsonld` is the default that's loaded if no config is passed, we may want to create a new file like `resources/full-example-config.jsonld` and then make `config.jsonld` a more representative default. Maybe something like this

```json
{
  "@context": {
    "@base": "https://ns.flur.ee/dev/config/main/",
    "@vocab": "https://ns.flur.ee/system#",
    "profiles": {
      "@container": [
        "@graph",
        "@id"
      ]
    }
  },
  "@id": "standaloneServer",
  "@graph": [
    {
      "@id": "localDiskStorage",
      "@type": "Storage",
      "addressIdentifier": "ee.flur/main",
      "filePath": "/opt/fluree-server/data"
    },
    {
      "@id": "connection",
      "@type": "Connection",
      "parallelism": 4,
      "cacheMaxMb": 200,
      "commitStorage": {
        "@id": "localDiskStorage"
      },
      "indexStorage": {
        "@id": "localDiskStorage"
      },
      "primaryPublisher": {
        "@type": "Publisher",
        "storage": {
          "@id": "localDiskStorage"
        }
      }
    },
    {
      "@id": "consensus",
      "@type": "Consensus",
      "consensusProtocol": "standalone",
      "maxPendingTxns": 42,
      "connection": {
        "@id": "connection"
      }
    },
    {
      "@id": "http",
      "@type": "API",
      "httpPort": 8090,
      "maxTxnWaitMs": 120000
    }
  ]
}
```

Currently if you run `server` as an uberjar with no config option or env vars passed in, it fails w/ aws client connect errors, I'm assuming because the default `config.jsonld` includes those hypothetical examples

Reply:
> @zonotope Right now it seems `config.jsonld` is an example of what's possible (e.g. with the s3 & ipns publishers). But since `resources/config.jsonld` is the default that's loaded if no config is passed, we may want to create a new file like `resources/full-example-config.jsonld` and then make `config.jsonld` a more representative default. Maybe something like this
> 
> ```json
> {
>   "@context": {
>     "@base": "https://ns.flur.ee/dev/config/main/",
>     "@vocab": "https://ns.flur.ee/system#",
>     "profiles": {
>       "@container": [
>         "@graph",
>         "@id"
>       ]
>     }
>   },
>   "@id": "standaloneServer",
>   "@graph": [
>     {
>       "@id": "localDiskStorage",
>       "@type": "Storage",
>       "addressIdentifier": "ee.flur/main",
>       "filePath": "/opt/fluree-server/data"
>     },
>     {
>       "@id": "connection",
>       "@type": "Connection",
>       "parallelism": 4,
>       "cacheMaxMb": 200,
>       "commitStorage": {
>         "@id": "localDiskStorage"
>       },
>       "indexStorage": {
>         "@id": "localDiskStorage"
>       },
>       "primaryPublisher": {
>         "@type": "Publisher",
>         "storage": {
>           "@id": "localDiskStorage"
>         }
>       }
>     },
>     {
>       "@id": "consensus",
>       "@type": "Consensus",
>       "consensusProtocol": "standalone",
>       "maxPendingTxns": 42,
>       "connection": {
>         "@id": "connection"
>       }
>     },
>     {
>       "@id": "http",
>       "@type": "API",
>       "httpPort": 8090,
>       "maxTxnWaitMs": 120000
>     }
>   ]
> }
> ```
> 
> Currently if you run `server` as an uberjar with no config option or env vars passed in, it fails w/ aws client connect errors, I'm assuming because the default `config.jsonld` includes those hypothetical examples

Yes, we should definitely do this. I was planning to in the final cleanup push, but I hadn't had a chance to yet.

Reply:
We should also convert the raft example too.